model_analyzer/config/generate/perf_analyzer_config_generator.py

```diff
@@ -69,16 +77,17 @@ def __init__(self, cli_config, model_names, model_perf_analyzer_flags,
 
     def is_done(self):
         """ Returns true if this generator is done generating configs """
-        # yapf: disable
-        return       self._all_configs_returned()   \
-              or     self._last_results_erroneous() \
-              or not self._throughput_gain_valid()
-        #yapf: enable
+        return self._done_walking() or self._last_results_erroneous()
 
     def next_config(self):
         """ Returns the next generated config """
-        config = self._configs[self._curr_config_index]
-        self._curr_config_index += 1
+        if self._done_walking_concurrencies():
```

Comment:
@nv-braf I could use some guidance on state machine. It seems bad or unclean to step one thing before, and another thing after. Is there a proper way to do this?